bundles/org.openhab.binding.radiothermostat/src/main/java/org/openhab/binding/radiothermostat/internal/handler/RadioThermostatHandler.java

```diff
@@ -0,0 +1,600 @@
+
+            if (httpStatus != OK_200) {
+                throw new RadioThermostatHttpException("Thermostat HTTP response code was: " + httpStatus);
+            }
+            output = contentResponse.getContentAsString();
+        } catch (RadioThermostatHttpException | InterruptedException | TimeoutException | ExecutionException
+                | IllegalStateException e) {
+            logger.warn("Error executing thermostat command: {}, {}", postJson, e.getMessage());
+        }
+
+        return output;
+
+    }
+
+    /**
+     * Update a given channelId from the thermostat data
+     * 
+     * @param the channel id to be updated
+     * @param data
+     * @return
+     * @throws Exception
+     */
+    public static @Nullable Object getValue(String channelId, RadioThermostatData data) throws Exception {
+        switch (channelId) {
+            case TEMPERATURE:
+                if (!data.getThermostatData().getTemperature().equals(new Double(0))) {
+                    return new QuantityType<Temperature>(data.getThermostatData().getTemperature(),
+                            API_TEMPERATURE_UNIT);
+                } else {
+                    return null;
+                }
+            case HUMIDITY:
+                if (data.getHumidity() != 0) {
+                    return new QuantityType<>(data.getHumidity(), API_HUMIDITY_UNIT);
+                } else {
+                    return null;
+                }
+            case MODE:
+                return data.getThermostatData().getMode();
+            case FAN_MODE:
+                return data.getThermostatData().getFanMode();
+            case PROGRAM_MODE:
+                return data.getThermostatData().getProgramMode();
+            case SET_POINT:
+                if (data.getThermostatData().getSetpoint() != 0) {
+                    return new QuantityType<Temperature>(data.getThermostatData().getSetpoint(), API_TEMPERATURE_UNIT);
+                } else {
+                    return null;
+                }
+            case OVERRIDE:
+                return data.getThermostatData().getOverride();
+            case HOLD:
+                if (data.getThermostatData().getHold() == 1) {
+                    return OnOffType.ON;
+                } else {
+                    return OnOffType.OFF;
+                }
+            case STATUS:
+                return data.getThermostatData().getStatus();
+            case FAN_STATUS:
+                return data.getThermostatData().getFanStatus();
+            case DAY:
+                return data.getThermostatData().getTime().getDayOfWeek();
+            case HOUR:
+                return data.getThermostatData().getTime().getHour();
+            case MINUTE:
+                return data.getThermostatData().getTime().getMinute();
+            case DATE_STAMP:
+                return data.getThermostatData().getTime().getThemostatDateTime();
+            case LAST_UPDATE:
+                return ZonedDateTime.now();
```

Comment:
Why are you always returning the current time? You shouldn't do this since a channel might be trying to refresh this value from a cached RadioThermostatData.

Reply:
I think I did this as a hold over from the rules way of doing things. It's a clue to know if your thermostat connection is dead. It also displays in the official app to tell you the last time the thermostat phoned home. But I'll take it out since the binding will report offline if contact is lost.